rfc/0010-cbft.md

```diff
@@ -138,38 +136,55 @@ Compound Queries:
 
 Some queries are more limited and usually used for debugging purposes:
 
- - **TermQuery:** A query that looks for **exact** matches of the term in the index (no analyzer, no stemming). Useful to check what the actual content of the index is. Usual better alternative is `MatchQuery`.
- - **Fuzzy Query:** A fuzzy query is a term query that matches terms within a specified edit distance (Levenshtein distance). Usual better alternative is `MatchQuery`.
+ - **TermQuery:** A query that looks for **exact** matches of the term in the index (no analyzer, no stemming). Useful to check what the actual content of the index is. It can also apply fuzziness on the term. Usual better alternative is `MatchQuery`.
  - **Prefix Query:** The prefix query finds documents containing terms that start with the provided prefix. Usual better alternative is `MatchQuery`.
  - **PhraseQuery:** A query that looks for **exact** match of several terms (in the exact order) in the index. Usual better alternative is `MatchPhraseQuery`.
  - **MatchAllQuery:** A query that matches all indexed documents.
  - **MatchNoneQuery:** A query that matches nothing.
 
 ## 2. High Level API Overview
-This section describes how the overall API looks like (to be expanded).
+This section describes how the overall API looks like.
+
+The search service is an HTTP endpoint that accepts `POST` requests. The url of the endpoint determine the FTS index to be queried, while the body of the request contains a `JSON` representation of the FTS query to perform:
+
+> **POST** /api/index/`{indexName}`/query
 
 For both bucket and cluster level, the approach is the same. Right now only Bucket level is defined in this RFC.
 
 ```java
-bucket.query(SearchQuery[, SearchParams]);
+bucket.query(String indexName, SearchQuery query[, SearchParams params]);
 ```
```

Comment:
Why not make indexName a property of SearchQuery? It shortens the signature and makes it a tighter API IMO since SearchQuery takes in the responsibility for the index to query.